search/utils.py

```diff
@@ -0,0 +1,31 @@
+import struct
+
+from django.db import connection
+
+
+def create_index_table():
+    cursor = connection.cursor()
+    cursor.execute("DROP TABLE IF EXISTS idx")
+    cursor.execute("CREATE VIRTUAL TABLE idx using "
+                   "FTS4(id, model, model_id, public, text)")
+
+
+def rank(match_info):
+    # Handle match_info called w/default args 'pcx' - based on the example
+    # rank function http://sqlite.org/fts3.html#appendix_a
+    # From github.com/coleifer/peewee/master/playhouse/sqlite_ext.py
```

Comment:
I know the following code is taken from somewhere else, but it would help to have some more comments in the following lines to understand what it's doing. It's a bit cryptic, as is.